src/Authentication/Authentication/Helpers/AuthenticationHelpers.cs

```diff
@@ -46,37 +46,24 @@ internal static void Logout(IAuthContext authConfig)
         {
             lock (FileLock)
             {
-                if (authConfig.AuthType == AuthenticationType.Delegated)
-                    File.Delete(Path.Combine(Constants.TokenCacheDirectory, Constants.UserCacheFileName));
-                else
-                    File.Delete(Path.Combine(Constants.TokenCacheDirectory, Constants.AppCacheFileName));
+                TokenCacheStorage.DeleteToken(authConfig.ClientId);
             }
         }
 
-        private static void ConfigureTokenCache(ITokenCache tokenCache, string tokenCacheFile)
+        private static void ConfigureTokenCache(ITokenCache tokenCache, string appId)
         {
-            if (!Directory.Exists(Constants.TokenCacheDirectory))
-                Directory.CreateDirectory(Constants.TokenCacheDirectory);
-
-            string tokenCacheFilePath = Path.Combine(Constants.TokenCacheDirectory, tokenCacheFile);
-
             tokenCache.SetBeforeAccess((TokenCacheNotificationArgs args) => {
                 lock (FileLock)
                 {
-                    args.TokenCache.DeserializeMsalV3(File.Exists(tokenCacheFilePath)
-                        ? TokenCryptographer.DecryptToken(File.ReadAllBytes(tokenCacheFilePath))
-                        : null,
-                        shouldClearExistingCache: true);
+                    args.TokenCache.DeserializeMsalV3(TokenCacheStorage.GetToken(appId), shouldClearExistingCache: true);
```

Comment:
>TokenCacheStorage.GetToken(appId) [](start = 54, length = 33)

is it safe to call Deserialize if this returns nothing?

Reply:
It's safe to return pass a null to `DeserializeMsalV3`. [MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/1cef43d185bb0a1dcc013d2ec232ae9d4c6c053a/src/client/Microsoft.Identity.Client/TokenCache.Serialization.cs#L42) does the check.